src/utilities.hpp

```diff
@@ -1,72 +0,0 @@
-#pragma once
-
-#include <chrono>
-#include <iostream>
-#include <string>
-#include <chrono>
-#include <thread>
-#include <cmath>
-#include <climits>
-#include <random>
-
-
-class MillisecondClock {
```

Comment:
I moved MillisecondClock, cliHelper, and myRandom to their own compilation units. These types had disparate dependencies and were also used in different places, so it didn't make sense to group them together. Also as a general best practice, it's best to have one class per file.

There's exceptions to that, but those really only come up when the class is not meant to be used outside of the compilation unit. This is one of those "rules" that it's best to know & follow for better code organization now, and later you'll maybe come across some situation where it's appropriate to break that rule.

Reply:
Originally I had them all separated in their own files like you showed but I combined them into a single utility file cause I thought it would be cleaner and save on comp time if there were less objects to create. If that is not the case, happy to run with the way you and originally I had it set up.